doc: Jade dependencies missing from test/README

With just the specific dependencies above (rather than everything in the Dockerfile), the qemu build fails for me on Ubuntu 21.10:

```
Dependencies
                      SDL support: NO
                SDL image support: NO
                      GTK support: NO
                           pixman: YES
                      VTE support: NO
                    slirp support: internal
                         libtasn1: YES
                              PAM: NO
                    iconv support: NO
                   curses support: NO
                    virgl support: NO
                     curl support: NO
                Multipath support: NO
                      VNC support: NO
                   brlapi support: NO
                      vde support: NO
                   netmap support: NO
                Linux AIO support: NO
           Linux io_uring support: NO
               ATTR/XATTR support: YES
                     RDMA support: NO
                   PVRDMA support: NO
                      fdt support: internal
                libcap-ng support: NO
                      bpf support: NO
                    spice support: NO
                      rbd support: NO
                   xfsctl support: NO
                smartcard support: NO
                      U2F support: NO
                           libusb: NO
                    usb net redir: NO
                   OpenGL support: NO
                              GBM: NO
                 libiscsi support: NO
                   libnfs support: NO
                  seccomp support: NO
                GlusterFS support: NO
                      TPM support: NO
                   libssh support: NO
                      lzo support: NO
                   snappy support: NO
                    bzip2 support: NO
                    lzfse support: NO
                     zstd support: NO
                NUMA host support: NO
                          libxml2: YES
                         capstone: NO
                  libpmem support: NO
                libdaxctl support: NO
                          libudev: YES
                       FUSE lseek: NO

  Subprojects
                    libvhost-user: YES

Found ninja-1.10.1 at /usr/bin/ninja
+ git checkout fd85235d17cd8813d6a31f5ced3c5acbf1933718
HEAD is now at fd85235 hw/xtensa: add ESP32 machine
+ git submodule update --recursive --init
+ ninja -C build
ninja: Entering directory `build'
[43/2098] Generating qemu-version.h with a meson_exe.py custom command
fatal: No names found, cannot describe anything.
[937/2098] Compiling C object libcommon.fa.p/disas_i386.c.o
FAILED: libcommon.fa.p/disas_i386.c.o 
cc -Ilibcommon.fa.p -I../dtc/libfdt -I../slirp -I../slirp/src -I. -Iqapi -Itrace -Iui -Iui/shader -I/usr/include/libmount -I/usr/include/blkid -I/usr/include/glib-2.0 -I/usr/lib/x86_64-linux-gnu/glib-2.0/include -I/usr/include/gio-unix-2.0 -I/usr/include/pixman-1 -fdiagnostics-color=auto -pipe -Wall -Winvalid-pch -Werror -std=gnu11 -O2 -g -isystem /home/dev/HWI/test/work/jade/qemu/linux-headers -isystem linux-headers -iquote . -iquote /home/dev/HWI/test/work/jade/qemu -iquote /home/dev/HWI/test/work/jade/qemu/include -iquote /home/dev/HWI/test/work/jade/qemu/disas/libvixl -iquote /home/dev/HWI/test/work/jade/qemu/tcg/i386 -pthread -fsanitize=undefined -fsanitize=address -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=2 -m64 -mcx16 -D_GNU_SOURCE -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE -Wstrict-prototypes -Wredundant-decls -Wundef -Wwrite-strings -Wmissing-prototypes -fno-strict-aliasing -fno-common -fwrapv -Wold-style-declaration -Wold-style-definition -Wtype-limits -Wformat-security -Wformat-y2k -Winit-self -Wignored-qualifiers -Wempty-body -Wnested-externs -Wendif-labels -Wexpansion-to-defined -Wimplicit-fallthrough=2 -Wno-missing-include-dirs -Wno-shift-negative-value -Wno-psabi -fstack-protector-strong -fPIC -D_REENTRANT -MD -MQ libcommon.fa.p/disas_i386.c.o -MF libcommon.fa.p/disas_i386.c.o.d -o libcommon.fa.p/disas_i386.c.o -c ../disas/i386.c
In file included from /usr/include/string.h:519,
                 from /home/dev/HWI/test/work/jade/qemu/include/qemu/osdep.h:87,
                 from ../disas/i386.c:34:
In function ‘strcpy’,
    inlined from ‘PNI_Fixup’ at ../disas/i386.c:6434:4,
    inlined from ‘PNI_Fixup’ at ../disas/i386.c:6400:1:
/usr/include/x86_64-linux-gnu/bits/string_fortified.h:79:10: error: ‘__builtin_memcpy’ offset [0, 7] is out of the bounds [0, 0] [-Werror=array-bounds]
   79 |   return __builtin___strcpy_chk (__dest, __src, __glibc_objsize (__dest));
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In function ‘strcpy’,
    inlined from ‘PNI_Fixup’ at ../disas/i386.c:6427:4,
    inlined from ‘PNI_Fixup’ at ../disas/i386.c:6400:1:
/usr/include/x86_64-linux-gnu/bits/string_fortified.h:79:10: error: ‘__builtin_memcpy’ offset [0, 5] is out of the bounds [0, 0] [-Werror=array-bounds]
   79 |   return __builtin___strcpy_chk (__dest, __src, __glibc_objsize (__dest));
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
cc1: all warnings being treated as errors
[944/2098] Compiling C object libcommon.fa.p/hw_scsi_mptsas.c.o
ninja: build stopped: subcommand failed.
```

--- a/test/README.md
+++ b/test/README.md
@@ -17,18 +17,21 @@ It also tests usage with `bitcoind`.
 - `test_coldcard.py` tests the command line interface and Coldcard implementation.
 It uses the [Coldcard simulator](https://github.com/Coldcard/firmware/tree/master/unix#coldcard-desktop-simulator).
 It also tests usage with `bitcoind`.
+- `test_jade.py` tests the command line interface and Blockstream Jade implementation.
+It uses the [Espressif fork of the Qemu emulator](https://github.com/espressif/qemu.git).
+It also tests usage with `bitcoind`.
 
-`setup_environment.sh` will build the Trezor emulator, the Coldcard simulator, the Keepkey emulator, the Digital Bitbox simulator, and `bitcoind`.
+`setup_environment.sh` will build the Trezor emulator, the Coldcard simulator, the Keepkey emulator, the Digital Bitbox simulator, the Jade emulator, and `bitcoind`.
 if run in the `test/` directory, these will be built in `work/test/trezor-firmware`, `work/test/firmware`, `work/test/keepkey-firmware`, `work/test/mcu`, and `work/test/bitcoin` respectively.
 In order to build each simulator/emulator, you will need to use command line arguments.
-These are `--trezor-1`, `--trezor-t`, `--coldcard`, `--keepkey`, `--bitbox01`, and `--bitcoind`.
+These are `--trezor-1`, `--trezor-t`, `--coldcard`, `--keepkey`, `--bitbox01`, `--jade`, and `--bitcoind`.
 If an environment variable is not present or not set, then the simulator/emulator or bitcoind that it guards will not be built.
 
-`run_tests.py` runs the tests. If run from the `test/` directory, it will be able to find the Trezor emulator, Coldcard simulator, Keepkey emulator, Digital Bitbox simulator, and bitcoind.
+`run_tests.py` runs the tests. If run from the `test/` directory, it will be able to find the Trezor emulator, Coldcard simulator, Keepkey emulator, Digital Bitbox simulator, Jade emulator, and bitcoind.
 Otherwise the paths to those will need to be specified on the command line.
-`test_trezor.py`, `test_coldcard.py`, `test_keepkey.py`, and `test/test_digitalbitbox.py` can be disabled.
+`test_trezor.py`, `test_coldcard.py`, `test_keepkey.py`, `test_jade.py`, and `test/test_digitalbitbox.py` can be disabled.
 
-If you are building the Trezor emulator, the Coldcard simulator, the Keepkey emulator, the Digital Bitbox simulator, and `bitcoind` without `setup_environment.sh`, then you will need to make `work/` inside of `test/`.
+If you are building the Trezor emulator, the Coldcard simulator, the Keepkey emulator, the Jade emulator, the Digital Bitbox simulator, and `bitcoind` without `setup_environment.sh`, then you will need to make `work/` inside of `test/`.
 
 ```
 $ cd test
@@ -162,6 +165,116 @@ Build the emulator:
 $ export PATH=$PATH:`pwd`/nanopb/generator
 $ cmake -C cmake/caches/emulator.cmake . -DNANOPB_DIR=nanopb/ -DKK_HAVE_STRLCAT=OFF -DKK_HAVE_STRLCPY=OFF
 $ make kkemu
+```
+
+## Jade emulator
+
+### Dependencies
+
+In order to build the Jade emulator, the following packages will need to be installed:
+
+```
+build-essential git cmake ninja-build libusb-1.0-0 libglib2.0-dev libpixman-1-dev libgcrypt20-dev
+```
+
+### Building
+
+Building the jade firmware and emulator can be a bit involved.  See `setup_environment.sh`.
+
+NOTE: the branch and commit of the esp-idf toolchain and the qemu emulator required are best extracted
+from the Jade Dockerfile at the Jade commit being built.
+
+Clone the jade repository and extract the branches and commits of the dependencies:
+
+```
+$ mkdir jade
+$ git clone --recursive --branch master https://github.com/Blockstream/Jade.git ./jade
+$ ESP_IDF_BRANCH=$(grep "ARG ESP_IDF_BRANCH=" Dockerfile | cut -d\= -f2)
+$ ESP_IDF_COMMIT=$(grep "ARG ESP_IDF_COMMIT=" Dockerfile | cut -d\= -f2)
+$ ESP_QEMU_BRANCH=$(grep "ARG ESP_QEMU_BRANCH=" Dockerfile | cut -d\= -f2)
+$ ESP_QEMU_COMMIT=$(grep "ARG ESP_QEMU_COMMIT=" Dockerfile | cut -d\= -f2)
+```
+
+Clone and build the qemu emulator:
+```
+$ mkdir qemu
+$ git clone --depth 1 --branch ${ESP_QEMU_BRANCH} --single-branch --recursive https://github.com/espressif/qemu.git ./qemu
+$ cd qemu && checkout ${ESP_QEMU_COMMIT}
+$ ./configure \
+    --target-list=xtensa-softmmu \
+    --enable-gcrypt \
+    --enable-sanitizers \
+    --disable-user \
+    --disable-opengl \
+    --disable-curses \
+    --disable-capstone \
+    --disable-vnc \
+    --disable-parallels \
+    --disable-qed \
+    --disable-vvfat \
+    --disable-vdi \
+    --disable-qcow1 \
+    --disable-dmg \
+    --disable-cloop \
+    --disable-bochs \
+    --disable-replication \
+    --disable-live-block-migration \
+    --disable-keyring \
+    --disable-containers \
+    --disable-docs \
+    --disable-libssh \
+    --disable-xen \
+    --disable-tools \
+    --disable-zlib-test \
+    --disable-sdl \
+    --disable-gtk \
+    --disable-vhost-scsi \
+    --disable-qom-cast-debug \
+    --disable-tpm
+$ ninja -C build
+$ cd ..
+```
+
+Clone and install the relevant version of the esp-idf libraries and toolchain:
+```
+$ mkdir ./esp && cd ./esp
+$ export IDF_TOOLS_PATH="$(pwd)/esp-idf-tools"
+$ git clone --quiet --depth=1 --branch ${ESP_IDF_BRANCH} --single-branch --recursive https://github.com/espressif/esp-idf.git
+$ cd esp-idf && git checkout ${ESP_IDF_COMMIT}
+$ ./install.sh esp32
+$ . ./export.sh
+$ cd ../..
+```
+(Note: once the tools are installed, any new shell only needs to source the `./export.sh` file.)
+
+Build the Jade fw configured for the emulator:
+```
+$ cd jade
+$ rm -f sdkconfig
+$ cp configs/sdkconfig_qemu.defaults sdkconfig.defaults
+$ idf.py all
+```
+
+Create an emulator rom image:
+```
+$ esptool.py --chip esp32 merge_bin --fill-flash-size 4MB -o main/qemu/flash_image.bin \
+$ --flash_mode dio --flash_freq 40m --flash_size 4MB \
+$ 0x9000 build/partition_table/partition-table.bin \
+$ 0xe000 build/ota_data_initial.bin \
+$ 0x1000 build/bootloader/bootloader.bin \
+$ 0x10000 build/jade.bin
+$ cd ..
+```
+
+Extract the minimal artifacts required to run the emulator
+```
+$ rm -fr simulator
+$ mkdir simulator
+$ cp qemu/build/qemu-system-xtensa simulator/
+$ cp -R qemu/pc-bios simulator/
+$ cp jade/main/qemu/flash_image.bin simulator/
+$ cp jade/main/qemu/qemu_efuse.bin simulator/
+$ cd ..
 ```
 
 ## Bitcoin Core

